### PR TITLE
feat: Implement Infinite-scroll on user profile

### PR DESCRIPTION
**Pull request by kunal00000**
issue: #591 

Initial number of photos -> 20
number of photos added on scroll -> 9
max number of photos -> 100 

Screenshot 2023-02-16 at 10 57 06 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
     "react-dropzone": "^14.2.1",
     "react-hook-form": "^7.34.2",
     "react-hotkeys-hook": "^3.4.7",
+    "react-icons": "^4.7.1",
```
Comment (vnugent) [inline package.json]: Is there a specific icon you're looking for?  we already have heroicon package installed

File: src/components/media/UserGallery.tsx [hunk 2]
```diff
@@ -120,6 +121,34 @@ export default function UserGallery ({ uid, postId: initialPostId, auth, userPro
...
+  useEffect(() => {
+    // add new items from the imageList to the imageListToShow array when length changes
+    if (imageList !== undefined) setImageListToShow(imageList.slice(0, imageListToShowLength))
+    console.log('Length Before ', imageListToShow.length)
+  }, [imageList, imageListToShowLength])
+
+  useEffect(() => {
+    console.log('Length After ', imageListToShowLength)
+
+    const handleScroll = (): void => {
```
Comment (vnugent) [inline src/components/media/UserGallery.tsx]: Sorry I forgot to mention in our chat on Discord that we're currently using `react-use` library that provides a wealth of useful hooks.   You can use the `useIntersection()` hook (built on Intersection Observer API) to simplify the scroll detection code.  It'd be more performant and less custom code :)

- Docs: https://streamich.github.io/react-use/?path=/story/sensors-useintersection--docs
- StickyHeader component https://github.com/OpenBeta/open-tacos/blob/develop/src/components/crag/StickyHeader.tsx

In StickyHeader variable `atTop` becomes true when the `<div ref={intersectionRef} ...>` reaches the top of the screen.   You can change `rootMargin` to be notified when you reach the bottom of the screen (I don't remember exactly how on top of my head).

**Review by vnugent — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kunal00000) [reply package.json]: I needed a Loader icon which I didn't find in heroicons. I checked it before using the react icons library.

Comment (vnugent) [reply package.json]: what about hero icon `arrow-path` with Tailwind `animate-spin` to create the loading effect?

Comment (kunal00000) [reply package.json]: Screenshot 2023-02-17 at 3 54 44 PM

It looks good. Let me revert the last commit.

File: src/components/media/UserGallery.tsx (hunk 2 above)
Comment (kunal00000) [reply src/components/media/UserGallery.tsx]: I tried this but not working, maybe I'll have to dive deeper into the working of `useIntersection()` hook (learning `react`).
But, I solved the problem of custom code. I implemented this again using `react-infinite-scroll-component` library. It was quite simple to use that which I didn't expect before.

**Comment by vnugent** [comment]
@kunal00000 
Thanks for the update.  Can you view the Preview build https://open-tacos-git-fork-kunal00000-infinite-scroll-openbeta-dev.vercel.app/u/viet-test-profile1 and let me know if the gallery is jittering or is it just my computer?

https://user-images.githubusercontent.com/3805254/220174994-0f1d9992-0450-4e82-92aa-8c981824adc3.mov

File: package.json [hunk 3]
```diff
@@ -54,6 +54,8 @@
...
```
Comment (vnugent) [inline package.json]: Please yarn remove this package.

**Comment by kunal00000** [comment]
@vnugent , Thanks for reviewing, 
Yes, it is jittering. would figure out the problem and update accordingly. 
If you have some suggestions please let me know.

**Comment by kunal00000** [comment]
The reason behind jittering is `rotate` property in `animate-spin`. 
Corners of icon intersect with the viewBox because of which height of viewBox increase and decrease on loading.

https://user-images.githubusercontent.com/92316166/220189113-2e54f93a-8286-423b-a131-21482265478f.mov

----
What else we can do is instead of loader an down arrow with `animate-bounce`.

https://user-images.githubusercontent.com/92316166/220189450-23a25971-1e75-40d9-94e5-d40f05b9667a.mov

**Pull request merged**